pkg/cmd/templates/templater.go

```diff
@@ -58,38 +59,61 @@ func UseOptionsTemplates(cmd *cobra.Command) {
 	cmd.SetUsageFunc(templater.UsageFunc())
 }
 
-func rootCmd(c *cobra.Command) []string {
-	root := []string{}
+func rootCmdNames(c *cobra.Command) []string {
+	cmds := rootCmds(c)
+	rootCmdNames := []string{}
+	for _, cmd := range cmds {
+		rootCmdNames = append(rootCmdNames, cmd.Name())
+	}
+	return rootCmdNames
+}
 
-	var appendCmdName func(*cobra.Command)
-	appendCmdName = func(x *cobra.Command) {
+func rootCmds(c *cobra.Command) []*cobra.Command {
+	root := []*cobra.Command{}
+
+	var appendCmd func(*cobra.Command)
+	appendCmd = func(x *cobra.Command) {
 		if x.HasParent() {
-			appendCmdName(x.Parent())
-			root = append(root, x.Parent().Name())
+			appendCmd(x.Parent())
+			root = append(root, x.Parent())
 		}
 	}
-	appendCmdName(c)
+	appendCmd(c)
 
-	if cName := c.Name(); c.HasSubCommands() && len(root) == 1 && root[0] == "openshift" && cName != "openshift" {
-		root = append(root, cName)
+	if c.HasSubCommands() && len(root) == 1 && root[0].Name() == "openshift" && c.Name() != "openshift" {
```

Comment:
This is now no longer safe (at all) because the master binary is going to change to be dual role.  We're going to have to make this be an argument to the generator.